src/services/EcoBridge/Socket/SocketBridge.ts

```diff
@@ -0,0 +1,584 @@
+
+        if (!activeChainId || !this._account) return
+
+        const allowance = await ApprovalsAPI.approveControllerFetchApprovals(
+          {
+            chainID: activeChainId.toString(),
+            owner: this._account,
+            allowanceTarget: approvalData.allowanceTarget.toString(),
+            tokenAddress: approvalData.approvalTokenAddress.toString()
+          },
+          { signal: this.renewAbortController('allowance') }
+        )
+
+        const {
+          result: { value },
+          success
+        } = allowance
+
+        if (success) {
+          if (BigNumber.from(value).lt(BigNumber.from(approvalData.minimumApprovalAmount))) {
+            this.store.dispatch(
+              this.actions.setApprovalData({
+                chainId: activeChainId ? activeChainId : 1,
+                allowanceTarget: approvalData.allowanceTarget,
+                tokenAddress: approvalData.approvalTokenAddress,
+                amount: approvalData.minimumApprovalAmount,
+                owner: approvalData.owner
+              })
+            )
+
+            this.store.dispatch(
+              ecoBridgeUIActions.setStatusButton({
+                label: 'Approve',
+                isLoading: false,
+                isError: false,
+                isApproved: false,
+                isBalanceSufficient: true
+              })
+            )
+            return
+          }
+
+          this.store.dispatch(
+            ecoBridgeUIActions.setStatusButton({
+              label: 'Bridge',
+              isLoading: false,
+              isError: false,
+              isApproved: true,
+              isBalanceSufficient: true
+            })
+          )
+        } else {
+          this.store.dispatch(
+            ecoBridgeUIActions.setStatusButton({
+              label: 'Something went wrong',
+              isLoading: false,
+              isError: true,
+              isApproved: false,
+              isBalanceSufficient: false
+            })
+          )
+        }
+      }
+    } catch (e) {
+      this.store.dispatch(
+        ecoBridgeUIActions.setBridgeModalStatus({ status: BridgeModalStatus.ERROR, error: getErrorMsg(e) })
+      )
+    }
+  }
+
+  public getBridgingMetadata = async () => {
+    const requestId = this.store.getState().ecoBridge[this.bridgeId as SocketList].lastMetadataCt
+
+    const helperRequestId = (requestId ?? 0) + 1
+
+    this.store.dispatch(this.actions.requestStarted({ id: helperRequestId }))
+
+    this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.LOADING }))
+
+    const { from, to } = this.store.getState().ecoBridge.ui
+
+    if (!from.chainId || !to.chainId || !this._account || !from.address || Number(from.value) === 0) return
+
+    const socketTokens = this.store.getState().ecoBridge.socket.lists[this.bridgeId]
+    const fromNativeCurrency = Currency.getNative(from.chainId)
+
+    let fromTokenAddress = ''
+    let toTokenAddress = ''
```

Comment:
does this declaration make any difference?
if not, then we can simply have something like
`let fromTokenAddress : string`

Reply:
`Variable 'fromTokenAddress' is used before being assigned.`